lib/facter/consul.rb

```diff
@@ -0,0 +1,24 @@
+require 'facter'
+
+# Example:
+# $ consul -v
+# Consul v0.5.0
+# Consul Protocol: 2 (Understands back to: 1)
+consul_v = Facter::Util::Resolution.exec("consul -v")
```

Comment:
I think just normal strings is ok.

I'm mostly concerned about guarding against consul not being available, string parsing, and timeouts. I've been bit by all those things.

There is another thing I would like to suggest in this refactor and that is to make it so it returns "unknown" when not available. This makes the fact easier to use in contexts when strict variables is enabled in puppet.

We do not currently test with strict variables on, but I think we should.